### PR TITLE
11 shopping cart

### LINKED ISSUE #11: Shopping Cart
Background: Items, and a user that is not logged in

As a visitor
When I visit any page with an item on it
I should see a link or button for "Add to Cart"
When I click "Add to cart" for that item
And I click a link or button to view cart
And my current path should be "/cart"
And I should see a small image, title, description and price for the item I just added
And there should be a "total" price for the cart that should be the sum of all items in the cart

When I login
And I visit "/cart"
Then I should see all of the data that was there when I was not logged in

### PR DESCRIPTION
**Pull request by applegrain**
- implement the shopping cart 
- add feature tests and model tests related to implementation of the cart
- fixed merge conflicts that arose when I merged Jai's and Jason's changes to the views 
- extracted #sign_in, #fill_in_form and #register_user to spec/spec_helper.rb 

closes #11

### CODE REVIEW

File: app/models/cart_item.rb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+class CartItem < SimpleDelegator
+  attr_reader :quantity
+  def initialize(item, quantity=0)
+    super(item)
+    @item = item
```
Comment (noobjey) [inline app/models/cart_item.rb]: What is this line doing? If the explanation will take to long we can just talk about it tomorrow

**Comment by noobjey** [comment]
I left a couple comments. Nothing big. Let me know if you want to merge.

**Comment by applegrain** [comment]
@noobjey I forgot to add increase/decrease button functionality so I'll do it now, take a look at your comments and then it should be good to merge.

**Comment by mrjaimisra** [comment]
dope dope dope! (as Drew would say)

I was thinking one thing when reading your cart tests, and this related to the session#destroy action in the sessions controller. since we are setting session[:id] = nil when we Sign Out, will this also clear the cart? I wonder if we can write a sad path test to determine whether or not a user can logout and still view his or her cart. If we want to address that scenario, I can think of a few creative ways to approach it...

overall awesome code Lovisa. Crushing it.

**Comment by applegrain** [comment]
@mrjaimisra good point, we would want to test something like that. And I agree about the contexts! I can change them

**Pull request merged**